spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateSpecTest.java

```diff
@@ -0,0 +1,634 @@
+ */
+
+package org.candlepin.spec;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertNotFound;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+
+import org.candlepin.dto.api.client.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.client.v1.AttributeDTO;
+import org.candlepin.dto.api.client.v1.CertificateDTO;
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.EntitlementDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.dto.api.client.v1.SubscriptionDTO;
+import org.candlepin.invoker.client.ApiException;;
+import org.candlepin.resource.HostedTestApi;
+import org.candlepin.resource.client.v1.EntitlementsApi;
+import org.candlepin.resource.client.v1.OwnerContentApi;
+import org.candlepin.resource.client.v1.OwnerProductApi;
+import org.candlepin.spec.bootstrap.assertions.OnlyInHosted;
+import org.candlepin.spec.bootstrap.assertions.OnlyInStandalone;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.client.api.ConsumerClient;
+import org.candlepin.spec.bootstrap.client.api.JobsClient;
+import org.candlepin.spec.bootstrap.client.api.OwnerClient;
+import org.candlepin.spec.bootstrap.client.cert.X509Cert;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.builder.Subscriptions;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.time.ZoneId;
+import java.util.List;
+
+
+@SpecTest
+public class EntitlementCertificateSpecTest {
+
+    private static ApiClient client;
+    private static OwnerDTO owner;
+    private OwnerClient ownerApi;
+    private EntitlementsApi entitlementsApi;
+    private OwnerContentApi ownerContentApi;
+    private OwnerProductApi ownerProductApi;
+    private ConsumerClient consumerApi;
+    private HostedTestApi hostedTestApi;
+    private JobsClient jobApi;
+    private ProductDTO monitoring;
+    private PoolDTO pool;
+    private SubscriptionDTO subscription;
+    private ConsumerDTO system;
+
+    @BeforeAll
+    public void beforeAll() throws ApiException {
```

Comment:
The class members didn't need to be static, so I made them non-static.

Reply:
It will have to be static once Ondrej's PR goes in because he's changing the `TestInstance.Lifecycle` (https://github.com/candlepin/candlepin/pull/3557/files#diff-a79a06da6cfad80f38cf42205b904d5f2ee7f37048f03bba288f0114c34da756). So depending on which PR goes in first, the other will have to adjust theirs

Reply:
I've merged Ondrej's PR, so this one needs changing because it's failing with `BeforeAll method 'public void org.candlepin.spec.EntitlementCertificateSpecTest.beforeAll() throws org.candlepin.invoker.client.ApiException' must be static`